client/src/components/Users/index.jsx

```diff
@@ -1,9 +1,237 @@
-import React from 'react'
-
-export default function Members() {
-  return (
-    <div>
-      Members
-    </div>
-  )
+import React, { Component } from 'react';
+import { Table, Icon, Dropdown } from 'antd';
+import Editable from 'react-contenteditable';
+import 'antd/dist/antd.css';
+import './style.css';
+import { filter, sort, deleteSwal, roleFilter } from './helpers.js';
+import Select from './select';
+import UserMenu from './menu/menu';
+import Form from './form/index';
+import Error from './error/Error'
+
+const users = require('../../utils/users.json');
+
+class Users extends Component {
+  state = {
+    users: [],
+    nameOptions: [],
+    showSaveButton: false,
+    userNameError: false,
+    fullNameError: false,
+    password: null,
+    passwordError: false,
+    rowSelected: null,
+    newRow: {},
+    saving: false,
+    saved: false,
+    show: false,
+  };
+
+  componentDidMount() {
+    //Fetch to get users from database//
+    //Store them in the state
+    this.setState({
+      users: [...users, { id: users.length + 1, user_name: '', full_name: '', role: 'developer' }],
+    });
+  }
+  componentDidUpdate(prevProps, prevState) {
+    const { users } = this.state;
+    if (prevState.users !== users) {
+      const allUsers = users.slice(0);
+      const nameOptions = filter(allUsers);
+      this.setState({ nameOptions });
+    }
+  }
+  handleAddUser = (event, columnName) => {
+    const newValue = event.target.value;
+    this.setState(prevState => {
+      const clonedUsers = JSON.parse(JSON.stringify(prevState.users));
+      clonedUsers[clonedUsers.length - 1][columnName] = newValue;
+      //Not sure if I should update the state now or after inserting the user in database!
+      return { users: clonedUsers, newRow: clonedUsers[clonedUsers.length - 1], showSaveButton: true };
+    });
+  };
+
+  saveNewUser = () => {
+    if (this.validateUserInfo(this.state.newRow)) {
+      if (this.state.password) {
+        //fetch to add user in the database
+        //change message
+        this.setState({ passwordError: false, saved: true });
+      } else {
+        this.setState({ passwordError: true });
+      }
+    }
+  };
+
+  handleEditUserInfo = (event, record, columnName) => {
+    if (record.id === this.state.users.length) {
```

Comment:
You can't rely on `record.id` that has to do anything with the `users` indexes
at some point, it should be different.

Reply:
I will change it to the following:
```javascript
const {users} = this.state;
if(record.id === users[users.length].id)```